ciphers/caesar_cipher.cpp

```diff
@@ -0,0 +1,123 @@
+/**
+ * @file caesar_cipher.cpp
+ * @brief Implementation of [Caesar cipher](https://en.wikipedia.org/wiki/Caesar_cipher) algorithm.
+ *
+ * @details
+ *      In cryptography, a Caesar cipher, also known as Caesar's cipher, the shift cipher, 
+ *      Caesar's code or Caesar shift, is one of the simplest and most widely known encryption 
+ *      techniques. It is a type of substitution cipher in which each letter in the plaintext 
+ *      is replaced by a letter some fixed number of positions down the alphabet. For example, 
+ *      with a left shift of 3, D would be replaced by A, E would become B, and so on. 
+ *      The method is named after Julius Caesar, who used it in his private correspondence.
+ *
+ * Algorithm -      
+ *      
+ *      The encryption can also be represented using modular arithmetic by first transforming 
+ *      the letters into numbers, according to the scheme, A → 0, B → 1, ..., Z → 25.
+ *      Encryption of a letter x by a shift n can be described mathematically as,
+ * 
+ *      E(x) = (x + n) mod 26
+ * 
+ *      while decryption can be described as,
+ * 
+ *      D(x) = (x - n) mod 26
+ * 
+ * \note This program implements caesar cipher for only uppercase english latters (i.e. A-Z). 
+ * 
+ * @author [Deep Raval](https://github.com/imdeep2905)
+ */
+#include <iostream>
+#include <string>
+#include <cassert>
+
+/** \namespace ciphers
+ * \brief Algorithms for encryption and decryption
+ */
+namespace ciphers {
+    namespace caesar {
+        /**
+         * This function finds character for given value (i.e.A-Z)
+         * @param x value for which we want character 
+         * @returns  corresponding character for perticular value
+         */        
+        char get_char(const int x) {
```

Comment:
consider adding `get_char` and `get_value` to an un-named namespace within `ciphers::caesar` so that these functions are hidden from the end-user. Also, they can be declared inline.